### PR TITLE
Various Jenkins-related tweaks.

### PR DESCRIPTION
**Pull request by nicolasnoble**
-) created a .profile at the root of the docker image - seems the new nvm script requires that.
-) installing simplejson and mako python packages, for the sanity check.
-) calling prepare_travis.sh in order to try and download precompiled packages for openssl and protobuf.
-) using the new "config" axis.
-) improved the output of prepare_travis.sh.

### CODE REVIEW

**Comment by nicolasnoble** [comment]
Also, this changes a tiny bit the script for Travis, so make sure travis is green if you merge that.

**Comment by nicolasnoble** [comment]
. . . and that didn't trigger a Jenkins build.

**Comment by jtattermusch** [comment]
Python error on Travis is unrelated, merging.

**Comment by jtattermusch** [comment]
Hm, i think I broke a lot of test by merging this. For some reason I didn't think of spawning the pull request build first. :-(

**Comment by jtattermusch** [comment]
retest this please

File: tools/jenkins/run_jenkins.sh [hunk 1]
```diff
@@ -56,9 +56,11 @@ then
     $FETCH_PULL_REQUEST_CMD \
     && git checkout -f $GIT_COMMIT \
     && git submodule update \
+    && pip install simplejson mako \
```
Comment (jtattermusch) [inline tools/jenkins/run_jenkins.sh]: eventually, we should move pip install into the Dockerfile.

**Comment by jtattermusch** [comment]
ssl_lib.c:188:1: error: missing initializer [-Werror=missing-field-initializers]
ssl_lib.c:188:1: error: (near initialization for 'ssl3_undef_enc_method.enc_flags') [-Werror=missing-field-initializers]
cc1: all warnings being treated as errors
make[2]: *** [ssl_lib.o] Error 1
make[2]: Leaving directory `/var/local/git/grpc/third_party/openssl/ssl'
make[1]: *** [build_ssl] Error 1
make[1]: Leaving directory `/var/local/git/grpc/third_party/openssl'
make: *** [libs/dbg/openssl/libssl.a] Error 2

**Pull request merged**